resources/listen.rb

```diff
@@ -22,8 +22,18 @@
       variables['listen'][new_resource.name] ||= {}
       variables['listen'][new_resource.name]['mode'] ||= ''
       variables['listen'][new_resource.name]['mode'] << new_resource.mode
-      variables['listen'][new_resource.name]['bind'] ||= ''
-      variables['listen'][new_resource.name]['bind'] << new_resource.bind
+      variables['listen'][new_resource.name]['bind'] ||= []
+      if new_resource.bind.is_a? Hash
+        new_resource.bind.map do |addresses, ports|
+          addresses = Array(addresses) unless addresses.is_a? Array
+          ports = Array(ports) unless ports.is_a? Array
```

Comment:
See my comment in frontend about the `Array()` wrapping, same thing applies here.